### PR TITLE
Added a solution for laying out binary trees(2)

### PR DESCRIPTION
**Pull request by joelwilliamson**
This one spaces the nodes based on how far they are from the bottom of the tree, 2**(h-d-1)

### CODE REVIEW

File: site/learn/tutorials/99problems.md [hunk 1]
```diff
@@ -1585,10 +1585,36 @@ out the rules and write the corresponding OCaml function.
 Hint: On a given level, the horizontal distance between neighbouring
 nodes is constant.
 
-<!-- SOLUTION -->
+The tree shown is 
+```ocamltop
+let example_layout_tree =
```
Comment (Chris00) [inline site/learn/tutorials/99problems.md:1590]: You need not to redefine it, the ocamltop session last the entire page.

Comment (joelwilliamson) [inline site/learn/tutorials/99problems.md:1590]: Its a different tree.

Comment (Chris00) [inline site/learn/tutorials/99problems.md:1590]: You're right, I checked too fast.

File: site/learn/tutorials/99problems.md [hunk 2]
```diff
@@ -1585,10 +1585,36 @@ out the rules and write the corresponding OCaml function.
...
+            Node('u', Node('p', Empty, leaf 'q'), Empty))
+```
 
-```ocaml
-  (* solution pending *)
+SOLUTION
+
+> ```ocamltop
+> let layout_binary_tree_2 t =
+>       let tree_height = height t in
```
Comment (Chris00) [inline site/learn/tutorials/99problems.md]: `height` is not available in the environment of the page.

File: site/learn/tutorials/99problems.md [hunk 3]
```diff
@@ -1585,10 +1585,36 @@ out the rules and write the corresponding OCaml function.
...
+>       let rec find_missing_left depth = function
+>               | Empty -> tree_height - depth
```
Comment (Chris00) [inline site/learn/tutorials/99problems.md]: The indentation is too large.

File: site/learn/tutorials/99problems.md [hunk 4]
```diff
@@ -1585,10 +1585,36 @@ out the rules and write the corresponding OCaml function.
...
+                      leaf 'm'),
+            Node('u', Node('p', Empty, leaf 'q'), Empty))
+```
 
-```ocaml
-  (* solution pending *)
+SOLUTION
+
+> ```ocamltop
+> let layout_binary_tree_2 t =
```
Comment (Chris00) [inline site/learn/tutorials/99problems.md:1600]: Align the next line with the first `t`.

**Pull request merged**